src/mame/jaleco/jaleco_vj_pc.cpp

```diff
@@ -0,0 +1,166 @@
+// license:BSD-3-Clause
+// copyright-holders:windyfairy
+/*
+Hardware to run VJ Windows PC side software
+
+Motherboard is similar to EP-5BVPXB but says REV 1.3 which matches NMC-5VXC
+but the board used does not have the NMC markings like the NMC-5VXC.
+
+VIA 82C586B PCI ISA/IDE bridge
+VIA 82C585VPX
+Winbond W83877F IO Core Logic
+
+S3 Virge/DX Q5C2BB (not connected to a monitor)
+
+UPS is connected to COM1 port
+
+CD-ROM: Mitsumi CRMC-FX3210S(?)
+Floppy: Unidentified but machine has floppy drive
+
+
+TODO: None of the VIA chips are emulated in the newer PCI code so Windows 95 will
+ask you to install the new hardware every boot. There's an autoexec.bat that'll
+restore the registry and other configs back to the factory setup every boot so
+the hardware will never stay "installed" between boots.
+
+Keyboard and mouse should not enabled by default but it'd be impossible to install drivers currently.
+It's recommended to disable the keyboard (Input Settings > Keyboard Selection > AT Keyboard > Disabled)
+when actually playing the games because otherwise you'll be sending inputs to the Windows PC in the background.
+*/
+
+#include "emu.h"
+
+#include "machine/pci.h"
+#include "machine/pci-ide.h"
+#include "machine/i82439hx.h"
+#include "machine/i82439tx.h"
+#include "machine/i82371sb.h"
+#include "video/virge_pci.h"
+#include "bus/isa/isa_cards.h"
+#include "machine/fdc37c93x.h"
+
+#include "jaleco_vj_pc.h"
+#include "jaleco_vj_qtaro.h"
+#include "jaleco_vj_sound.h"
+#include "jaleco_vj_ups.h"
+
+
+jaleco_vj_pc_device::jaleco_vj_pc_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+    device_t(mconfig, JALECO_VJ_PC, tag, owner, clock),
+    m_maincpu(*this, "maincpu")
+{
+}
+
+void jaleco_vj_pc_device::device_start()
+{
+}
+
+void jaleco_vj_pc_device::device_reset()
+{
+}
+
+static void isa_internal_devices(device_slot_interface &device)
+{
+    device.option_add("fdc37c93x", FDC37C93X);
+}
+
+static void isa_com(device_slot_interface &device)
+{
+    // TODO: Stepping Stage uses a different UPS but will still boot unlike VJ without a valid UPS attached
+    device.option_add("jaleco_vj_ups", JALECO_VJ_UPS);
+}
+
+static void isa_cards(device_slot_interface &device)
+{
+    device.option_add("jaleco_vj_sound", JALECO_VJ_ISA16_SOUND);
+}
+
+void jaleco_vj_pc_device::superio_config(device_t *device)
+{
+    fdc37c93x_device &fdc = *downcast<fdc37c93x_device *>(device);
```

Comment:
The real VIA uses a Winbond W83877F Super I/O, please add a `// TODO` here

Reply:
Done. Also turned the other hardware-related comments into TODOs.